Blockzilla/Menu/Protocol/MenuItemProvider.swift

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+protocol MenuItemProvider: AnyObject {
+    var shortcutManager: ShortcutsManager { get set }
+    
+    func openInFireFoxItem(for url: URL) -> MenuAction?
+    func openInChromeItem(for url: URL) -> MenuAction?
+    func openInDefaultBrowserItem(for url: URL) -> MenuAction
+    
+    var findInPageItem: MenuAction { get }
+    var requestDesktopItem: MenuAction { get }
+    var requestMobileItem: MenuAction { get }
+    var settingsItem: MenuAction { get }
+    var helpItem: MenuAction { get }
+    var copyItem: MenuAction { get }
+    
+    func getShortcutsItem(for url: URL) -> MenuAction?
+    func addToShortcutsItem(for url: URL) -> MenuAction
+    func removeFromShortcutsItem(for url: URL) -> MenuAction
+    func sharePageItem(for utils: OpenUtils, sender: UIView) -> MenuAction
+}
+
+extension MenuItemProvider where Self: MenuActionable {
+    func openInFireFoxItem(for url: URL) -> MenuAction? {
+        canOpenInFirefox
+        
+        ? MenuAction(title: UIConstants.strings.shareOpenInFirefox, image: "open_in_firefox_icon") { [unowned self] in
+            self.openInFirefox(url: url)
+        }
+        : nil
+    }
+    
+    func openInChromeItem(for url: URL) -> MenuAction? {
+        canOpenInChrome ?
+        MenuAction(title: UIConstants.strings.shareOpenInChrome, image: "open_in_chrome_icon") { [unowned self] in
+            self.openInChrome(url: url)
+        }
+        : nil
+    }
+    
+    func openInDefaultBrowserItem(for url: URL) -> MenuAction {
+        MenuAction(title: UIConstants.strings.shareOpenInDefaultBrowser, image: "icon_favicon") { [unowned self] in
+            self.openInDefaultBrowser(url: url)
+        }
+    }
+    
+    var findInPageItem: MenuAction {
+        MenuAction(title: UIConstants.strings.shareMenuFindInPage, image: "icon_searchfor") { [unowned self] in
```

Comment:
For some reason these types of IDs `icon_searchfor` are not found now in the app tree and so the test fails to find the element: `FindInPageTest.testActivityMenuFindInPageAction()`